src/cpu/x64/injectors/jit_uni_binary_injector.cpp

```diff
@@ -2647,22 +2647,55 @@ template <typename T>
 typename std::enable_if<!(std::is_same<T, Xbyak::Zmm>::value
                           || std::is_same<T, Xbyak::Address>::value)>::type
 jit_uni_binary_injector_t<isa, Vmm>::execute_prelu_binary(const Vmm &dst,
-                                                        const Vmm &lhs, const T &rhs) const {
-    // in sse4 vmm_aux0 as mask it's index must be 0
-    Vmm vmm_aux0 = Vmm(rhs_arg_static_params_.rhs_prelu_helper_vmm_idx);
-
-    if (dst == vmm_aux0) {
-        vmm_aux0 = Vmm(14);
-        if (isa == sse41)
-            assert(!"conflict mask register");
-    }
+    const Vmm &lhs, const T &rhs) const {
+    if (is_superset(isa, avx)) {
+        host_->uni_vmulps(rhs, rhs, lhs);
+        host_->uni_vblendvps(dst, lhs, rhs, lhs);
+    } else {
+        using dnnl::impl::utils::one_of;
+        // in sse4 vmm_aux0 as mask it's index must be 0
+        Vmm vmm_aux0 = Vmm(rhs_arg_static_params_.rhs_prelu_helper_vmm_idx);
```

Comment:
The SSE implement assume that rhs_arg_static_params_.rhs_prelu_helper_vmm_idx = 0 and seems current binary_post_ops would set rhs_prelu_helper_vmm_idx = 0. However, Can we directly use vmm_aux0 = Vmm(0)  and not  use this rhs_prelu_helper_vmm_idx? Or add some check rhs_prelu_helper_vmm_idx must be 0, assert would not be compiled into release version. Maybe directly use  vmm_aux0 = Vmm(0) ? It would be easier to understand this piece of code.

Reply:
In theory this `rhs_arg_static_params_.rhs_prelu_helper_vmm_idx` should be a hint that helps to avoid the search for vacant register. Also since this is the SSE path, which is mostly supported as a legacy, I wouldn't put more effort on ironing out this part.